### PR TITLE
houdini: change sha256, add missing libs, fix license dir patch, fix …

### PR DESCRIPTION
**Pull request by antonxy**
…keyboard input

- Changed sha256 of src file, as this was apparantly changed on the
website

- Added missing libs: some X libs, alsa, dbus

- Changed patch to $out/houdini/Licensing.opt
localValidatorDir = ... was commented out in original file
so sed had no effect

- add export QT_XKB_CONFIG_ROOT
Without it only modifier keys worked, no text input

###### Motivation for this change

I wanted to try houdini today but it did not work out of the box. With these changes I got it working.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by joachifm** [comment]
cc @canndrew. I think the checksum change warrants some investigation.

File: pkgs/applications/misc/houdini/runtime.nix [hunk 1]
```diff
@@ -50,11 +58,13 @@ stdenv.mkDerivation rec {
                       --no-root-check \
                       --accept-EULA \
                       $out
-    sed -i "s|/usr/lib/sesi|${license_dir}|g" $out/houdini/Licensing.opt
+    echo -e "localValidatorDir = ${license_dir}\nlicensingMode = localValidator" > $out/houdini/Licensing.opt
```
Comment (Mic92) [inline pkgs/applications/misc/houdini/runtime.nix]: Is changing licensing scheme the reason they broke checksums? I am not sure how insightful a request towards support@sidefx.com would be to ask, if they plan to continue this practice in future.

Comment (canndrew) [inline pkgs/applications/misc/houdini/runtime.nix]: I don't think it's related. Making the license checker happy was the main chore in writing this package in the first place. I stopped fiddling with things as soon as it started working on my system, @antonxy is just removing a redundant hack that I left in and changing a config option that should have been changed. (I wonder why it worked without that change on my box though?)

**Comment by canndrew** [comment]
I just went to download the installer from their website and it looks like the minor version number has changed (633 => 671). Which would explain the checksum failure. @antonxy could you change that in the PR aswell?

**Comment by joachifm** [comment]
Are they serving 671 at the 633 url?

**Comment by canndrew** [comment]
There isn't a URL in the package. You have to go to their website, accept their EULA and then it starts downloading. I guess @antonxy could have found a link to download an old version through their website, but I'm guessing they just didn't change the version number.

**Comment by antonxy** [comment]
I did download the 633 version from the production builds section. I could of course get the current version, check if it works and update it.

**Comment by canndrew** [comment]
> I did download the 633 version from the production builds section
Oh, sorry. Weird that the hash changed then :/

**Comment by Mic92** [comment]
Merged without testing since the download requires registration.

**Pull request merged**